src/cluster_asm.c

```diff
@@ -1711,13 +1692,13 @@ void asmCron(void) {
     asmTask *task = listNodeValue(listFirst(asmManager->tasks));
 
     if (task->operation == ASM_IMPORT) {
-        if (task->state == ASM_FAILED) {
+        if (task->state == ASM_FAILED && task->state != ASM_CANCELED) {
```

Comment:
task->state only has a single value.

for canceled tasks, i think we can move them into done_tasks